textacy/texts.py

```diff
@@ -1,63 +1,86 @@
+            by spacy before assignment to the ``TextDoc.spacy_doc`` attribute
+        spacy_pipeline (``spacy.<lang>.<Lang>()``, optional): if a spacy pipeline
+            has already been loaded or used to make the input ``spacy.Doc``, pass
+            it in here to speed things up a touch; in general, only one of these
+            pipelines should be loaded per process
+        lang (str, optional): if doc's language is known, give its 2-letter code
+            (https://cloud.google.com/translate/v2/using_rest#language-params);
+            if None (default), lang will be automatically inferred from text
+        metadata (dict, optional): dictionary of relevant information about the
+            input ``text_or_sdoc``, e.g.::
+
+                {'title': 'A Search for Second-generation Leptoquarks in pp Collisions at √s = 7 TeV with the ATLAS Detector',
+                 'author': 'Burton DeWilde', 'pub_date': '2012-08-01'}
     """
-    def __init__(self, text, spacy_pipeline=None, lang='auto',
-                 metadata=None, max_cachesize=5):
+    def __init__(self, text_or_sdoc, spacy_pipeline=None, lang=None, metadata=None):
         self.metadata = {} if metadata is None else metadata
-        self.lang = text_utils.detect_language(text) if lang == 'auto' else lang
-        if spacy_pipeline is None:
-            self.spacy_pipeline = data.load_spacy_pipeline(lang=self.lang)
-        else:
-            # check for match between text and supplied spacy pipeline language
-            if spacy_pipeline.lang != self.lang:
-                msg = 'TextDoc.lang {} != spacy_pipeline.lang {}'.format(
-                    self.lang, spacy_pipeline.lang)
-                raise ValueError(msg)
-            else:
-                self.spacy_pipeline = spacy_pipeline
-        self.spacy_vocab = self.spacy_pipeline.vocab
-        self.spacy_stringstore = self.spacy_vocab.strings
-        self.spacy_doc = self.spacy_pipeline(text)
         self._term_counts = Counter()
-        self._cache = LRUCache(maxsize=max_cachesize)
+
+        if isinstance(text_or_sdoc, str):
+            self.lang = text_utils.detect_language(text_or_sdoc) if not lang else lang
+            if spacy_pipeline is None:
+                spacy_pipeline = data.load_spacy_pipeline(lang=self.lang)
+            # check for match between text and passed spacy_pipeline language
+            else:
+                if spacy_pipeline.lang != self.lang:
+                    msg = 'TextDoc.lang {} != spacy_pipeline.lang {}'.format(
+                        self.lang, spacy_pipeline.lang)
+                    raise ValueError(msg)
+            self.spacy_vocab = spacy_pipeline.vocab
+            self.spacy_stringstore = self.spacy_vocab.strings
+            self.spacy_doc = spacy_pipeline(text_or_sdoc)
+
+        elif isinstance(text_or_sdoc, sdoc):
+            self.lang = spacy_pipeline.lang if spacy_pipeline is not None else \
```

Comment:
@danvalente Okay, I've come full circle on this. I think I'd rather assume that users are instantiating a TextDoc with a correctly processed spaCy Doc, since double-checking their work isn't computationally "free".